Template sets grpc, opencensus, perfmark, errorprone

moljac opened the pull request:
### Describe your contribution

1. Moved `Google.Grpc` to this repo from:
   https://github.com/xamarin/XamarinComponents/tree/main/XPlat/Google.Grpc
2. Moved `Xamarin.Io.OpenCensus.*` to this repo from: 
  https://github.com/xamarin/XamarinComponents/tree/main/Android/Io.OpenCensus
3. Moved `Xamarin.Io.PerfMark.PerfMarkApi` to this repo from:
  https://github.com/xamarin/XamarinComponents/tree/main/Android/Io.PerfMark
4. addded new bindings
  - `com.google.errorprone:error_prone_annotations`
  - `com.google.android:annotations`
  - `org.codehaus.mojo:animal-sniffer-annotations`
  
6. created template sets


## Updated/new (moved) artifacts

- com.google.guava:guava - 28.2.0 -> 29.0-android
- com.squareup.okio:okio -  -> 1.13.0
- io.grpc:grpc-context -  -> 1.28.0
- io.grpc:grpc-core -  -> 1.28.0
- io.grpc:grpc-api -  -> 1.28.0
- io.opencensus:opencensus-api -  -> 0.28.3
- io.opencensus:opencensus-contrib-grpc-metrics -  -> 0.28.3
- io.perfmark:perfmark-api -  -> 0.23.0
- javax.inject:javax.inject - 1 -> 1.0.0
- com.github.bumptech.glide:glide -  -> 4.11.0
- com.google.errorprone:error_prone_annotations -  -> 2.3.4
- com.google.android:annotations -  -> 4.1.1.4
- org.codehaus.mojo:animal-sniffer-annotations -  -> 1.18
- com.google.android.gms:play-services-ads-lite -  -> 20.0.0


### Does this change any of the generated binding API's?

`net6-android` TFM added

### Artifacts updated



Checklist
- [x] Add any metadata needed to get PR to compile
- [x] Verify any new namespaces are properly renamed (utilities.cake)
- [x] Verify nuget/namespace spelling (utilities.cake)

jpobst commented on config.json:
```diff
@@ -1369,31 +1377,89 @@
+        "artifactId": "perfmark-api",
+        "version": "0.23.0",
+        "nugetVersion": "0.23.0",
+        "nugetId": "Xamarin.Io.OpenCensus.OpenCensusContribGrpcMetrics",
```
Incorrect NuGet Id.

moljac replied:
fixed Xamarin.Io.PerfMark.PerfMarkApi

jpobst commented on source/GooglePlayServicesProject.cshtml:
```diff
@@ -146,14 +146,20 @@
   <ItemGroup>
     @foreach (var art in @Model.MavenArtifacts)
     {
+      if (@Model.NuGetPackageId == "Xamarin.Firebase.ProtoliteWellKnownTypes" )
```
I don't think this is related to this PR?

moljac replied:
the whole `JavaSourceJar` and `JavaDocJar` is currently commented out. I left it because tooling has issues with few packages and commented part is actually list of problematic packages (preparing issue for that).

jpobst reviewed (commented):
The api-diffs on this are too big to review.  I think we need to take some steps to reduce them first:

- Publish NuGets for the `main` branch to get rid of previous diffs like CroNet and Firebase stuff.
- This repo should not bind `GoogleGson` since we've already added it to AndroidX.
- The artifacts being moved into this repo should match the most recent versions on NuGet.
  -  GRPC: 1.28 -> 1.37

We can manage with this PR, but I think in the future we need to keep these PR's smaller to make them more reviewable.

moljac commented:
grpc 1.3x. errors:

```
Dependency errors : 7
1
	System.Exception: Artifact 'io.grpc.grpc-protobuf-lite' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
2
	System.Exception: Artifact 'io.grpc.grpc-stub' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
3
	System.Exception: Artifact 'io.grpc.grpc-okhttp' has unknown 'Runtime' dependency 'com.squareup.okio.okio'. Either fulfill or exclude this dependency.
4
	System.Exception: Artifact 'io.grpc.grpc-okhttp' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
5
	System.Exception: Artifact 'io.grpc.grpc-core' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
6
	System.Exception: Artifact 'io.grpc.grpc-android' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
7
	System.Exception: Artifact 'io.grpc.grpc-api' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.

```

moljac commented:
> The api-diffs on this are too big to review. I think we need to take some steps to reduce them first:
> 
> * Publish NuGets for the `main` branch to get rid of previous diffs like CroNet and Firebase stuff.

  Done!
  
> * This repo should not bind `GoogleGson` since we've already added it to AndroidX.

  removed!
  
> * The artifacts being moved into this repo should match the most recent versions on NuGet.
>   
>   * GRPC: 1.28 -> 1.37

  This was legacy (dependencyOnly = true) was not bumped for a while. Bumping to 1.3X (started with 1.37) but could not resolve dependencies.
   
> We can manage with this PR, but I think in the future we need to keep these PR's smaller to make them more reviewable.

Unfortunate start template picked which "pulled in" other dependencies as template sets (and even new artifacts).

jpobst commented:
> We can manage with this PR, but I think in the future we need to keep these PR's smaller to make them more reviewable.

I was wrong, there are still way too many diffs to review.  My theory would be that there would not be many/any changes since we are simply moving repos, but there's still plenty of changes.

Let's break each template set into a new PR.  We can start at the bottom (that is, whichever set does not depend on any other sets), get it committed, and then work on the next one.

moljac commented:
<PackageReference Include="Xamarin.Google.Dagger" Version="2.27.0" />

> I was wrong, there are still way too many diffs to review. My theory would be that there would not be many/any changes since we are simply moving repos, but there's still plenty of changes.

`grpc` was unfortunate pick by me. Sorry. And it simply kept on pulling in other stuff.

Also, some of the bindings from XC did cut the corner[s] with `ReferenceJar` or `InputJar`, so no bindings.

> Let's break each template set into a new PR. We can start at the bottom (that is, whichever set does not depend on any other sets), get it committed, and then work on the next one.

I'll need to check which were the last in this transitive chain, but I'll do it tomorrow. To tired right now
